pkg/diff/chunk.go

```diff
@@ -17,260 +17,532 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
-	"reflect"
+	"strings"
 
 	"github.com/pingcap/errors"
 	"github.com/pingcap/parser/model"
 	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/pingcap/tidb-tools/pkg/utils"
 	log "github.com/sirupsen/logrus"
 )
 
+var (
+	equal string = "="
+	lt    string = "<"
+	lte   string = "<="
+	gt    string = ">"
+	gte   string = ">="
+)
+
+type bound struct {
+	column      string
+	lower       string
+	lowerSymbol string
+	upper       string
+	upperSymbol string
+}
+
 // chunkRange represents chunk range
 type chunkRange struct {
-	begin interface{}
-	end   interface{}
-	// for example:
-	// containBegin and containEnd is true, means [begin, end]
-	// containBegin is true, containEnd is false, means [begin, end)
-	containBegin bool
-	containEnd   bool
-
-	// if noBegin is true, means there is no lower limit
-	// if noEnd is true, means there is no upper limit
-	noBegin bool
-	noEnd   bool
+	bounds []*bound
 }
 
-// CheckJob is the struct of job for check
-type CheckJob struct {
-	Schema string
-	Table  string
-	Column *model.ColumnInfo
-	Where  string
-	Args   []interface{}
-	Chunk  chunkRange
+// newChunkRange return a chunkRange, if boundNum is greater than 0, will creates a chunkRange with fixed bound num.
+func newChunkRange(boundNum int) *chunkRange {
+	var bounds []*bound
+	if boundNum > 0 {
+		bounds = make([]*bound, boundNum)
```

Comment:
this `boundNum > 0` only be used by `chunkRange.copy` right? if so, I think we can put this logic into `copy` directly because returning a slice with meaningless elements is dangerous.

Reply:
done